src/createPersistor.js

```diff
@@ -52,17 +50,41 @@ export default function createPersistor (store, config) {
           return
         }
 
-        let key = storesToProcess[0]
-        let storageKey = createStorageKey(key)
-        let endState = transforms.reduce((subState, transformer) => transformer.in(subState, key), stateGetter(store.getState(), key))
-        if (typeof endState !== 'undefined') storage.setItem(storageKey, serializer(endState), warnIfSetError(key))
+        persistCurrentStateForKey(state, storesToProcess[0])
         storesToProcess.shift()
       }, debounce)
     }
 
     lastState = state
   })
 
+  function findStoresToProcess (state) {
+    let keys = []
+    stateIterator(state, (subState, key) => {
+      if (!passWhitelistBlacklist(key)) return
+      let newState = stateGetter(state, key)
+      if (stateGetter(lastState, key) === newState) return
+      keys.push(key)
+    })
+
+    return keys
+  }
+
+  function persistCurrentStateForKey (state, key) {
+    let storageKey = createStorageKey(key)
+    let currentState = stateGetter(state, key)
+    let endState = transforms.reduce((subState, transformer) => transformer.in(subState, key), currentState)
+    if (typeof endState === 'undefined') return null
+    return storage.setItem(storageKey, serializer(endState), warnIfSetError(key))
+  }
+
+  function persistCurrentState () {
+    let state = store.getState()
+    let promises = storesToProcess.map(k => persistCurrentStateForKey(state, k))
+    storesToProcess.splice(0)
+    return promises
```

Comment:
1. `promises` can contain `null` values because `persistCurrentStateForKey` can return nulls. Is it desired behaviour? What if user will do `Primise.all(persistor.flush()).then(...)`?

2. May be better api for user will be to return single `Promise` object?

May be do something like:
```js
const promises = storesToProcess.map(k => persistCurrentStateForKey(state, k))
return Promise.all(promises.filter(v => v != null))
```